tags.go

```diff
@@ -200,3 +206,66 @@ func (client *Client) DeleteTag(id ID) error {
 	err := client.Mutate(&m, v, WithName("TagDelete"))
 	return HandleErrors(err, m.Payload.Errors)
 }
+
+// ReconcileTags manages tags API operations for TaggableResourceInterface implementations
+//
+// Tags not in 'tagsWanted' will be deleted, new tags from 'tagsWanted' will be created. Reconciled tags are returned.
+func (client *Client) ReconcileTags(resourceType TaggableResourceInterface, tagsWanted []string) ([]Tag, error) {
+	var err error
+	var tagConnection *TagConnection
+	var assignedTags []Tag
+
+	// Get underlying resource type for type checking
+	switch resource := resourceType.(type) {
+	case *Domain:
+		tagConnection, err = resource.GetTags(client, nil)
+	case *InfrastructureResource:
+		tagConnection, err = resource.GetTags(client, nil)
+	case *Repository:
+		tagConnection, err = resource.GetTags(client, nil)
+	case *Service:
+		tagConnection, err = resource.GetTags(client, nil)
+	case *System:
+		tagConnection, err = resource.GetTags(client, nil)
+	case *Team:
+		tagConnection, err = resource.GetTags(client, nil)
+	case *User:
+		tagConnection, err = resource.GetTags(client, nil)
+	default:
+		return assignedTags, fmt.Errorf("not a taggable resource type: %s" + string(resourceType.ResourceType()))
+	}
+	if err != nil {
+		return assignedTags, err
+	}
+	if tagConnection == nil {
+		return assignedTags, fmt.Errorf("no tags found on %s with id '%s'", string(resourceType.ResourceType()), resourceType.ResourceId())
+	}
+
+	// delete tags found in resource but not listed in tagsWanted
+	for _, tag := range tagConnection.Nodes {
+		if !slices.Contains(tagsWanted, tag.Flatten()) {
+			if err := client.DeleteTag(tag.Id); err != nil {
+				return assignedTags, err
```

Comment:
Since this function performs multiple API calls if one delete call fails I don't think the entire function should stop.

The errors join can be used here (link to example) - https://pkg.go.dev/errors#Join

Reply:
Shouldn't the expectation here be that a delete tag call actually deletes each tag? What is the goal of bypassing errors to accumulate further errors?

Reply:
I think the case Taimoor is going after is this:

Say you have 5 tags to delete
1.) Success
2.) Failure <- in your code it stops here but if it didn't
3.) Success
4.) Failure
5.) Success <- If we accumulate the errors we can report the end user that 2 tags were unable to be deleted for N reasons.

Reply:
Yes exactly like what we do here in `client.CreateTags()` - https://github.com/OpsLevel/opslevel-go/blob/8016ef2dfc61c16bd04913d73eb651794e4d4d1e/tags.go#L149

Reply:
Since `ReconcileAliases(0` also exits on the first error and returns the first error we can ignore this for now.